canon/src/id.rs

```diff
@@ -132,14 +132,6 @@ impl Id {
             Ok(Some(Store::take_bytes(self)?))
         }
     }
-
-    // This is a conveniance function to be called from Repr, in order not to
-    // have to construct an Id to get the encoded_len correctly.
-    pub(crate) fn encoded_len_for_payload_len(payload_len: usize) -> usize {
-        let actual_payload = core::cmp::min(payload_len, PAYLOAD_BYTES);
-        // version, length and the actual payload length
-        1 + (payload_len as u32).encoded_len() + actual_payload
-    }
```

Comment:
Not sure why this function was removed since it's the same logic that is applied twice below.
I would suggest to keep it and just call it in both `encoded_len` and `Repr`.
I believe the bug was that you were calling `cmp::min` twice, both before `encoded_len_for_payload_len` and inside the function.

Reply:
Since `Repr` is deprecated and will be removed, i don't think having duplicate implementations is a problem, since one should go away anyway.